components/TextInput.js

```diff
@@ -1,35 +1,88 @@
-import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
+import autoBindReact from 'auto-bind/react';
+import PropTypes from 'prop-types';
 import { TextInput as RNTextInput } from 'react-native';
 
-import { connectStyle } from '@shoutem/theme';
 import { connectAnimation } from '@shoutem/animation';
+import { connectStyle } from '@shoutem/theme';
+
+import { Caption } from './Text';
+import { View } from './View';
 
 class TextInput extends PureComponent {
-  render() {
-    const { props } = this;
-    const style = {
-      ...props.style,
+  constructor(props) {
+    super(props);
+
+    autoBindReact(this);
+
+    this.state = {
+      isFocused: props.highlightWhenFocused && props.autoFocus,
```

Comment:
Careful with autoFocus prop here. If there's multiple text inputs with autoFocus, only one of them will be focused. Maybe make use of something like this to detect if the field is actually focused -> https://reactnative.dev/docs/textinput#isfocused .

Reply:
Also, let's go for a shorter prop name, something like highlightOnFocus

Reply:
Much like a regular React Native `TextInput`, it's up to the developer to make sure there's only one `TextInput` with `autofocus`.

Will change the name.

Reply:
`isFocused()` method requires refs which does not seem at all worth the implementation in my humble opinion.

Reply:
What's the problem with refs ? It shouldn't be that hard to implement. It would provide you with the exact and true information, that is immune to developer mistake. I believe that will only lead to a better code.